pkg/runtime/pubsub/default_bulkpub.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"context"
+	"errors"
+
+	"golang.org/x/sync/errgroup"
+
+	contribPubsub "github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/utils"
+)
+
+const (
+	bulkPublishSeriallyKey       string = "bulkPublishSerially"
+	bulkPublishMaxConcurrencyKey string = "bulkPublishMaxConcurrency"
+
+	defaultBulkPublishMaxConcurrency int = 100
+)
+
+var ErrBulkPublishFailure = errors.New("bulk publish failed")
+
+// defaultBulkPublisher is the default implementation of BulkPublisher.
+// It is used when the component does not implement BulkPublisher.
+type defaultBulkPublisher struct {
+	p contribPubsub.PubSub
+}
+
+// NewDefaultBulkPublisher returns a new defaultBulkPublisher from a PubSub.
+func NewDefaultBulkPublisher(p contribPubsub.PubSub) *defaultBulkPublisher {
+	return &defaultBulkPublisher{
+		p: p,
+	}
+}
+
+// BulkPublish publishes a list of messages to a topic as individual Publish requests.
+// If 'bulkPublishSerially' metadata is set to true, the messages are sent to the broker serially,
+// in the same order. Otherwise they are sent to the broker as parallel Publish requests.
+func (p *defaultBulkPublisher) BulkPublish(_ context.Context, req *contribPubsub.BulkPublishRequest) (contribPubsub.BulkPublishResponse, error) {
+	if req.Metadata[bulkPublishSeriallyKey] == "true" {
```

Comment:
Please use `utils.IsTruthy` which is case-insensitive and supports also `1`, `yes`, `y`, `t`, etc.

Reply:
Addressed via https://github.com/dapr/dapr/pull/5396